WordPress/Classes/Categories/UIImageView+Gravatar.m

```diff
@@ -50,19 +50,21 @@ - (void)setImageWithGravatarEmail:(NSString *)emailAddress fallbackImage:(UIImag
     }];
 }
 
-- (void)setImageWithBlavatarUrl:(NSString *)blavatarUrl
+- (void)setImageWithSiteIcon:(NSString *)siteIcon
 {
     UIImage *blavatarDefaultImage = [UIImage imageNamed:BlavatarDefault];
 
-    [self setImageWithBlavatarUrl:blavatarUrl placeholderImage:blavatarDefaultImage];
+    [self setImageWithSiteIcon:siteIcon placeholderImage:blavatarDefaultImage];
 }
 
-- (void)setImageWithBlavatarUrl:(NSString *)blavatarUrl placeholderImage:(UIImage *)placeholderImage
+- (void)setImageWithSiteIcon:(NSString *)siteIcon placeholderImage:(UIImage *)placeholderImage
 {
-    if ([blavatarUrl rangeOfString:@"gravatar.com/blavatar"].location == NSNotFound) {
-        [self setImageWithURL:[self blavatarURLForHost:blavatarUrl] placeholderImage:placeholderImage];
+    if ([self isPhotonURL:siteIcon]) {
+        [self setImageWithURL:[self siteIconURLForSiteIconUrl:siteIcon] placeholderImage:placeholderImage];
+    } else if ([self isBlavatarURL:siteIcon]) {
+        [self setImageWithURL:[self blavatarURLForBlavatarURL:siteIcon] placeholderImage:placeholderImage];
     } else {
-        [self setImageWithURL:[self blavatarURLForBlavatarURL:blavatarUrl] placeholderImage:placeholderImage];
+        [self setImageWithURL:[self blavatarURLForHost:siteIcon] placeholderImage:placeholderImage];
```

Comment:
Are you talking about this bit: https://github.com/wordpress-mobile/WordPress-iOS/blob/issue/3720-jetpack-blavatars/WordPress/Classes/ViewRelated/Reader/ReaderBrowseSiteViewController.m#L100-L104? If so, I don't think the REST response has it, I might be looking into the wrong place though: https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/posts/